src/java/org/jivesoftware/openfire/plugin/rest/utils/LoggingUtils.java

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ */
+
+package org.jivesoftware.openfire.plugin.rest.utils;
+
+import org.jivesoftware.openfire.plugin.rest.RESTServicePlugin;
+import org.jivesoftware.util.JiveGlobals;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+
+public class LoggingUtils {
+    private static final Logger AUDIT_LOG = LoggerFactory.getLogger("RestAPI-Plugin-Audit");
+
+    public enum AuditEvent {
+        //Clustering
+        CLUSTERING_GET_STATUS,
+        CLUSTERING_GET_NODES,
+        CLUSTERING_GET_NODE,
+
+        //Groups
+        GROUPS_LIST,
+        GROUPS_GET_BY_NAME,
+        GROUPS_CREATE,
+        GROUPS_UPDATE_BY_NAME,
+        GROUPS_DELETE,
+
+        //JustMarried
+        USER_CHANGE_NAME,
+
+        //Messages
+        MESSAGE_BROADCAST,
+
+        //Message Archive
+        MESSAGE_ARCHIVE_UNREAD_COUNT,
+
+        //MUC
+        // - MUC Affiliations
+        MUC_LIST_AFFILIATED_USERS_FOR_AFFILIATION,
+        MUC_REPLACE_AFFILIATED_USERS_FOR_AFFILIATION,
+        MUC_REMOVE_AFFILIATED_USER_OR_GROUP_FOR_AFFILIATION,
+        MUC_ADD_AFFILIATED_USERS_FOR_AFFILIATION,
+        MUC_ADD_AFFILIATED_USER_OR_GROUP_AS_ADMIN,
+        MUC_ADD_AFFILIATED_USER_OR_GROUP_AS_MEMBER,
+        MUC_ADD_AFFILIATED_USER_OR_GROUP_AS_OUTCAST,
+        MUC_ADD_AFFILIATED_USER_OR_GROUP_AS_OWNER,
+        // - MUC Rooms
+        MUC_LIST_ROOMS,
+        MUC_GET_ROOM,
+        MUC_DELETE_ROOM,
+        MUC_CREATE_ROOM,
+        MUC_UPDATE_ROOM,
+        MUC_GET_PARTICIPANT_LIST,
+        MUC_GET_OCCUPANT_LIST,
+        MUC_GET_ROOM_HISTORY,
+        MUC_INVITE_USER,
+        ;
+    }
+
+    public static void auditEvent(AuditEvent event){
+        auditEvent(event, "");
+    }
+
+    public static void auditEvent(AuditEvent event, Object... parameters){
+        if (JiveGlobals.getBooleanProperty(RESTServicePlugin.SERVICE_LOGGING_ENABLED, false)) {
+            String parameterString = parseParameters(parameters);
+            String logMessage = "Event: " + event;
+            logMessage += " - ";
+            logMessage += "Parameters: + " + parameterString;
+            logMessage += " - ";
+            logMessage += "Caller: " + getCaller();
+            AUDIT_LOG.info(logMessage);
+        };
+    }
+
+    private static String parseParameters(Object[] parameters) {
+        //TODO: Does this belong here?
+        ArrayList<String> parsed = new ArrayList<>();
+        for (Object obj: parameters) {
+            if(obj == null){
+                parsed.add("null");
+            } else {
+                try {
+                    parsed.add(obj.toString());
+                } catch (Exception e) {
+                    parsed.add("unparseable");
+                }
+            }
+        }
+        return parsed.toString();
+    }
+
+    /*
+    * Returns the name and method of the calling class.
+    */
+    private static String getCaller() {
```

Comment:
Wouldn't this be pretty static, for by far most calls? Does this add enough value for it to be added?

Reply:
It's a private method that I added because it makes `auditEvent` more readable.

Reply:
Did I misunderstand this comment? It's getting the class and method of the caller of the audit event, and so feels useful.

A thought though - with well named classes and methods, do we actually need those enum values?
Instead of `CLUSTER_GET_STATUS`, we'd have `ClusteringController::getClusterStatus`. Would that be enough for an enterprise log analysis system, given that they're pretty stable? Current implementation has exactly 1 usage of each enum.